src/main/java/jmgomez/apipolicy/config/RequestInterceptor.java

```diff
@@ -0,0 +1,62 @@
+package jmgomez.apipolicy.config;
+
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import jmgomez.apipolicy.service.PolicyService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+import org.springframework.web.method.HandlerMethod;
+import org.springframework.web.servlet.HandlerInterceptor;
+import org.springframework.web.servlet.HandlerMapping;
+
+import java.io.IOException;
+import java.util.Map;
+
+@Component
+public class RequestInterceptor implements HandlerInterceptor {
+
+    private final PolicyService policyService;
+
+    @Autowired
+    public RequestInterceptor(@Lazy PolicyService policyService) {
+        this.policyService = policyService;
+    }
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws ServletException, IOException {
+        String authUserId = SecurityContextHolder.getContext().getAuthentication().getName();
+
+        Map<String, String> variables = (Map<String, String>) request.getAttribute(HandlerMapping.URI_TEMPLATE_VARIABLES_ATTRIBUTE);
+
+        String userId = authUserId;
+        String policyId = variables.get("policyId");
+        String accidentId = variables.get("accidentId");
+
+        if(handler instanceof HandlerMethod) {
+            if(request.getRequestURI().equals("/policies") && userId != null) {
```

Comment:
if/else if hell... think about how messy could get if you have a few more endpoints... you only need to check policyId and accidentId... it would be overly simple to check the the variables you got. But maybe mixing it with some pattern matching...

I mean, you need to check the policy with a /policies/${policyId} base URL, for example. And this will cover all the descendant URLs